src/puppetlabs/jdbc_util/core.clj

```diff
@@ -159,3 +160,17 @@
                        [sql 0]
                        seq-params-w-indices)]
     (vec (conj (flatten parameters) sql'))))
+
+(defn wrap-with-db-permission-error-handling
```

Comment:
on names

#1 this is an old discussion but I think we used to talk about hating the "wrap-with" naming pattern. I wouldn't mind something like "handle-permission-errors" imo.
#2 given that this will be used next to RBAC middlewares I'd like to be super clear that this has nothing to do with RBAC permissions or RBAC's db layer specifically.. what if we said "...postgres-authz-errors.." or "postgres-perm-errors"?